pulsar-io/jdbc/src/main/java/org/apache/pulsar/io/jdbc/JdbcUtils.java

```
@@ -0,0 +1,174 @@
+
+import com.google.common.collect.Lists;
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.IntStream;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Jdbc Utils
+ */
+@Slf4j
+public class JdbcUtils {
+
+    @Data(staticConstructor = "of")
+    @Setter
+    @Getter
+    @EqualsAndHashCode
+    @ToString
+    public static class TableId {
+        private final String catalogName;
+        private final String schemaName;
+        private final String tableName;
+    }
+
+    @Data(staticConstructor = "of")
+    @Setter
+    @Getter
+    @EqualsAndHashCode
+    @ToString
+    public static class ColumnId {
+        private final TableId tableId;
+        private final String name;
+        // SQL type from java.sql.Types
+        private final int type;
+        private final String typeName;
+        // column position in table
+        private final int position;
+    }
+
+    @Data(staticConstructor = "of")
+    @Setter
+    @Getter
+    @EqualsAndHashCode
+    @ToString
+    public static class TableDefinition {
+        private final TableId tableId;
+        private final List<ColumnId> columns;
+    }
+
+    /**
+     * Given a driver type(such as mysql), return its jdbc driver class name.
+     * TODO: test and support more types
+     */
+    public static String getDriverClassName(String driver) throws Exception {
+        if (driver.equals("mysql")) {
+            return "com.mysql.jdbc.Driver";
+        } if (driver.equals("sqlite")) {
+            return "org.sqlite.JDBC";
+        } else {
+            throw new Exception("Not tested jdbc driver type: " + driver);
```

probably log a warning message rather than failing it?

---

Thanks, we may better tested before add more types, fail early is better. For new type, we should also add dependency in pom file.